### PR TITLE
Jasc code from web

### PR DESCRIPTION
**Pull request by joseangel-sc**

### CODE REVIEW

File: shader_source.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from dataclasses import dataclass
+
+import os
+import requests
+
+
+@dataclass
+class GetShaderCode:
+    shade_id: str
```
Comment (jayaarrgh) [inline shader_source.py]: rename shader_id

File: shader_source.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from dataclasses import dataclass
+
+import os
+import requests
+
+
+@dataclass
```
Comment (jayaarrgh) [inline shader_source.py]: Let's just use a plain ol' class for now. The init should set the api_key, but I think the shader_id should be passed into the `_request_shader(self, shader_id)` and the public `get_shader_codes(self, shader_id)`

File: shader_source.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+class GetShaderCode:
+    shade_id: str
+    api_key: str = os.environ['API_KEY']
+
+    def _request_shade(self):
```
Comment (jayaarrgh) [inline shader_source.py]: rename `_request_shader` and take `shader_id` as a parameter

File: shader_source.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        response = requests.get(f"https://www.shadertoy.com/api/v1/shaders/{self.shade_id}?key={self.api_key}")
+        return response
+
+    def get_shades(self):
```
Comment (jayaarrgh) [inline shader_source.py]: rename `get_shader_passes` and take `shader_id` as a parameter.

File: shader_source.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        response = self._request_shade()
+        if response.status_code != 200:
+            return None
+        shader_info = response.json()['Shader']['renderpass']
```
Comment (jayaarrgh) [inline shader_source.py]: could rename shader_info to renderpasses

File: shader_source.py (hunk 2 above)
Comment (joseangel-sc) [reply shader_source.py]: yeap, agree

File: shader_source.py (hunk 3 above)
Comment (joseangel-sc) [reply shader_source.py]: :facepalm: sure, that make way more sense than creating a new object everytime

File: shader_source.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import os
+import requests
+
+
+class GetShaderCode:
```
Comment (jayaarrgh) [inline shader_source.py]: Maybe we should rename this to ShadertoyAPI or Shadertoy.
Prefer noun classes with verb methods, so in that case even ShaderCodeGetter would be better even though thats icky too haha

File: shader_source.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        return response
+
+    def get_shades(self, shader_id):
+        response = self._request_shader(shader_id)
+        if response.status_code != 200:
+            return None
+        renderpasses = response.json()['Shader']['renderpass']
+        code = []
+        for renderpass in renderpasses:
+            code.append(renderpass['code'])
```
Comment (jayaarrgh) [inline shader_source.py:27]: Maybe we should `continue` on `renderpass['type'] != 'image'` - I don't think other types of passes are going to work, but maybe we just play with the data first.

Comment (joseangel-sc) [reply shader_source.py:27]: Sorry, dumb question, just want to make sure , here https://www.shadertoy.com/view/ldtGWj, we want `Buffer A` and not image, right?

**Pull request merged**